src/Security/RolesMatrixBuilder.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Security;
+
+use Psr\Log\LoggerInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+final class RolesMatrixBuilder implements RolesBuilderInterface
+{
+    public const ROLE_ADMIN = 'ROLE_ADMIN';
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var array
+     */
+    private $rolesHierarchy;
+
+    /**
+     * @var array
+     */
+    private $labelPermission = [];
+
+    /**
+     * @var array
+     */
+    private $labelAdmin = [];
+
+    /**
+     * @var array
+     */
+    private $exclude = [];
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(
+        TokenStorageInterface $tokenStorage,
+        AuthorizationCheckerInterface $authorizationChecker,
+        Pool $pool,
+        LoggerInterface $logger,
+        array $rolesHierarchy = []
+    ) {
+        $this->tokenStorage = $tokenStorage;
+        $this->authorizationChecker = $authorizationChecker;
+        $this->pool = $pool;
+        $this->rolesHierarchy = $rolesHierarchy;
+        $this->logger = $logger;
+    }
+
+    public function setTranslator(TranslatorInterface $translator): void
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * @param string|bool|null $domain
+     */
+    public function getRoles($domain = false, bool $expanded = true): array
+    {
+        $roles = [];
+
+        if (!$this->tokenStorage->getToken()) {
+            return $roles;
+        }
+
+        $this->iterateAdminRoles(function ($role, $isMaster) use ($domain, &$roles): void {
+            $roles[$role] = $this->translateRole($role, $domain);
+        });
+
+        $roleSuperAdmin = $this->pool->getOption('role_super_admin');
+        $roleSonataAdmin = $this->pool->getOption('role_admin');
+
+        $baseRoles = [
+            $roleSuperAdmin,
```

Comment:
as this is the only place it is used, could you remove `$roleSuperAdmin` and just call `$this->pool->getOption('role_super_admin')`